### PR TITLE
Update for 1.18.2

### PR DESCRIPTION
**Pull request by theCALiC0**
I noticed that the current version does not work on 1.18.2 due to the changes to tags. I am also submitting a PR to Dossier for this same reason. Since FabricToolTags was removed, I have instead used the `c:shears` tag for that purpose.

### CODE REVIEW

**Comment by theCALiC0** [comment]
PR for Dossier (required for this to work): https://github.com/TerraformersMC/Dossier/pull/3

**Review by jhbuchanan45 — APPROVED** [review]
@Jade-TheCat  Thanks for this and the changes to Dossier, I've been meaning to do it but been too busy to look at it until recently. 

Everything here also looks fine to me, I'll build it locally and have a quick test run later but can't see anything problematic. 

@NeusFear could you take a look at this and the linked Dossier changes when you get chance?

**Review by NeusFear — APPROVED** [review]
All looks good to me, once we get the changes we requested on dossier fixed we can re-run the tests and hopefully merge if all pass

File: src/main/java/com/terraformersmc/campanion/client/renderer/item/FakeWorld.java [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 import com.terraformersmc.campanion.item.PlaceableTentItem;
 import com.terraformersmc.campanion.mixin.AccessorBiomeAccess;
+
+import net.fabricmc.fabric.impl.registry.sync.FabricRegistry;
```
Comment (Wyn-Price) [inline src/main/java/com/terraformersmc/campanion/client/renderer/item/FakeWorld.java:6]: What's with all the import changes

File: src/main/java/com/terraformersmc/campanion/recipe/TarpRecipe.java [hunk 2]
```diff
@@ -26,9 +27,9 @@ public boolean matches(CraftingInventory inv, World world) {
 		int shearAmount = 0;
 		for (int i = 0; i < inv.size(); i++) {
 			ItemStack stack = inv.getStack(i);
-			if(FabricToolTags.SHEARS.contains(stack.getItem())) {
+			if(stack.isIn(CampanionItemTags.SHEARS)) {
```
Comment (Wyn-Price) [inline src/main/java/com/terraformersmc/campanion/recipe/TarpRecipe.java:30]: We don't need our own shears tag. Please use `net.fabricmc.fabric.api.tag.convention.v1.ConventionalItemTags.SHEARS`

File: src/main/java/com/terraformersmc/campanion/recipe/TarpRecipe.java [hunk 3]
```diff
@@ -48,7 +49,7 @@ public DefaultedList<Ingredient> getIngredients() {
 		list.add(Ingredient.fromTag(ItemTags.WOOL));
 		list.add(Ingredient.fromTag(ItemTags.WOOL));
 
-		list.add(Ingredient.fromTag(FabricToolTags.SHEARS));
+		list.add(Ingredient.fromTag(CampanionItemTags.SHEARS));
```
Comment (Wyn-Price) [inline src/main/java/com/terraformersmc/campanion/recipe/TarpRecipe.java:52]: Again, use the fabric provided sheers tag

File: src/main/java/com/terraformersmc/campanion/recipe/TarpRecipe.java [hunk 4]
```diff
@@ -77,12 +78,12 @@ public DefaultedList<ItemStack> getRemainder(CraftingInventory inventory) {
 		DefaultedList<ItemStack> defaultedList = DefaultedList.ofSize(inventory.size(), ItemStack.EMPTY);
 
 		for(int i = 0; i < defaultedList.size(); ++i) {
-			Item item = inventory.getStack(i).getItem();
+			ItemStack stack = inventory.getStack(i).copy();
```
Comment (Wyn-Price) [inline src/main/java/com/terraformersmc/campanion/recipe/TarpRecipe.java:81]: Is there a reason for the copy here?

File: src/main/java/com/terraformersmc/campanion/recipe/TarpRecipe.java [hunk 5]
```diff
@@ -77,12 +78,12 @@ public DefaultedList<ItemStack> getRemainder(CraftingInventory inventory) {
...
+			Item item = stack.getItem();
 			if (item.hasRecipeRemainder()) {
 				defaultedList.set(i, new ItemStack(item.getRecipeRemainder()));
 			}
-			if(FabricToolTags.SHEARS.contains(item)) {
-				ItemStack stack = inventory.getStack(i).copy();
+			if(stack.isIn(CampanionItemTags.SHEARS)) {
```
Comment (Wyn-Price) [inline src/main/java/com/terraformersmc/campanion/recipe/TarpRecipe.java:86]: Please use the fabric provided shears tag

File: src/main/java/com/terraformersmc/campanion/tag/CampanionItemTags.java [hunk 6]
```diff
@@ -1,32 +1,40 @@
...
-	public static final Tag.Identified<Item> MRE_COMPONENTS = get("mre_components");
+	public static final TagKey<Item> FRUITS = get("fruits");
+	public static final TagKey<Item> GRAINS = get("grains");
+	public static final TagKey<Item> PROTEINS = get("proteins");
+	public static final TagKey<Item> VEGETABLES = get("vegetables");
+	public static final TagKey<Item> MRE_COMPONENTS = get("mre_components");
 
-	private static Tag.Identified<Item> get(String id) {
-		return AccessorItemTags.callRegister(Campanion.MOD_ID + ":" + id);
+	public static final TagKey<Item> SHEARS = TagKey.of(Registry.ITEM_KEY, new Identifier("c", "shears"));
```
Comment (Wyn-Price) [inline src/main/java/com/terraformersmc/campanion/tag/CampanionItemTags.java:29]: We don't need to mirror the fabric shears tag

File: src/main/resources/data/c/tags/items/shears.json [hunk 7]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (Wyn-Price) [inline src/main/resources/data/c/tags/items/shears.json:1]: Please use the fabric provided shears tag

**Review by Wyn-Price — CHANGES_REQUESTED** [review]
Most of this is just mappings changes. 
You seem to have missed the fabric shears tag at `net.fabricmc.fabric.api.tag.convention.v1.ConventionalItemTags.SHEARS`, and while what is currently done *would* work, there is no need to mirror it.

**Comment by theCALiC0** [comment]
Closing since a 1.18.2 update has already gone through! Thank yall

**Pull request closed without merging**